src/login/pam_systemd.c

```diff
@@ -251,14 +253,12 @@ static int get_seat_from_display(const char *display, const char **seat, uint32_
         if (r < 0)
                 return r;
 
-        r = get_ctty(ucred.pid, NULL, &tty);
+        r = get_ctty_devnr(ucred.pid, &display_ctty);
         if (r < 0)
                 return r;
 
-        v = vtnr_from_tty(tty);
-        if (v < 0)
-                return v;
-        else if (v == 0)
+        v = minor(display_ctty);
+        if (v < 0 || v > 63)
                 return -ENOENT;
```

Comment:
hmm, this can only work if we know for sure the algorithm how the VT subsystem maps VTs to major/minor numbers. So far we tried to avoid hardcoding assumptions on major/minor allocations, hence I'd like to avoid to start here...

I see the benefit of avoid /dev, hence the purpose of the patch makes sense to me. But I wonder if we can't find a nicer way that avoids hardcoding major/minor number allocation assumptions.

What about just using /sys/dev/char/$major:$minor instead? i.e. readlink() that and take the basename of what it returns? i.e. consider:

```
ls -l /sys/dev/char/4:4
lrwxrwxrwx. 1 root root 0 28. Jan 15:37 /sys/dev/char/4:4 -> ../../devices/virtual/tty/tty4/
```

PrivateDevices= does not affect /sys, hence this should make possible what you are trying to do.

Reply:
Yes, that would work.